File: Python/First Unique Character in a String.py
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        dic = {}
        for ch in s:
            if ch in dic:
                dic[ch] += 1
            else:
                dic[ch] = 1
        for index, ch in enumerate(s):
            if dic[ch] == 1:
                return index
        else:
            return -1
            

File: cpp/First Unique Character in a String.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unsigned int counter[26];
        memset(counter, 0, sizeof(counter));
        for (char ch : s) {
            counter[ch - 'a']++;
        }
        for (int i = 0; i < s.size(); i++) {
            if (counter[s[i] - 'a'] == 1) {
                return i;
            }
        }
        return -1;
    }
};
